### PR TITLE
Generated the ros package using catkin 

### PR DESCRIPTION
**Pull request by airfield20**
I regenerated the package using catking and refactored the includes. We probably should set an include directory in cmake in the future. For now the includes are ../include/file.h. 

The pull does not build. The original didnt have any build system included so it did not build either. Ill fix the build issues in the next pull.

---
This change is [Reviewable](https://reviewable.io/reviews/iscumd/obstacle_detection/2)

### CODE REVIEW

**Comment by airfield20** [comment]
This will fix #1

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+)
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(
+# include
```
Comment (airfield20) [inline CMakeLists.txt]: Just testing githubs code review functionality

Comment (airfield20) [reply CMakeLists.txt]: Uncomment this

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+# add_message_files(
+#   FILES
+#   Message1.msg
+#   Message2.msg
+# )
+
+## Generate services in the 'srv' folder
+# add_service_files(
+#   FILES
+#   Service1.srv
```
Comment (airfield20) [inline CMakeLists.txt:59]: May need to use this in the future if we use a map service

**Review by davidcutting — CHANGES_REQUESTED** [review]
Thanks for fixing the ROS set-up. I totally forgot about that. Here are my thoughts on your other changes:

- Why make our own Point2D? ROS has one built in. Functions which are a part of the point class should be members of Obstacle.

- Instead of "position()," use "get_position()" to match the "set_position()." As a general note, we should probably agree on our naming conventions and commenting requirements. Ex: "getPosition()" or "get_position()."

- I prefer separating obstacletype and obstacle into their own files. One file per header.

- RectangleBoundary should just be a type of obstacle, because it uses the same format as an obstacle anyway. Making it part of it's own class is redundant.

- Renaming the files to reflect the class name would make them easier to find and include.

**Comment by airfield20** [comment]
`Why make our own Point2D? ROS has one built in. Functions which are a part of the point class should be members of Obstacle.`
Using the ros point makes all of code dependent on our libraries dependent on ros. If we decide to switch to another communications marshaling library it will require a rewrite.


`Instead of "position()," use "get_position()" to match the "set_position()." As a general note, we should probably agree on our naming conventions and commenting requirements. Ex: "getPosition()" or "get_position()."`
I don't have an issue with this. removing get from every function name makes it easier to use autocomplete, change if you want. I dont mind, just be consistent across all of the files, in this node.


`I prefer separating obstacletype and obstacle into their own files. One file per header.` 
If you look at STL or boost, things directly related to each other are put into the same header file. Ex: std::unordered_set and std::set are in the same header. Again I dont mind if you change it.

 `RectangleBoundary should just be a type of obstacle, because it uses the same format as an obstacle anyway. Making it part of it's own class is redundant.`
RectangleBoundary shouldnt be a type of obstacle, we should actually also have a boundary type be typedef as a std::vector<ISC::Point2D> to represent a polygon that defines our course. These can be included in the same header. A boundary isnt an obstacle because its not tangible. Its not a physical object you are avoiding but rather a software defined point that shouldnt be exceeded. These two things will be handled differently in code, thus should be separate.

`Renaming the files to reflect the class name would make them easier to find and include.` 
Go ahead and change, not a problem to me.

File: include/boundary.h [hunk 3]
```diff
@@ -1,12 +0,0 @@
-#ifndef BOUNDARY_H
-#define BOUNDARY_H
-
-#include "geometry_msgs/Point.h"
```
Comment (airfield20) [inline include/boundary.h:4]: This makes our code ros dependent. Lets avoid doing this

File: src/obstacle_detection.cpp [hunk 4]
```diff
@@ -1,43 +0,0 @@
-#include "geometry_msgs/Point.h"
-#include "obstacle.h"
-#include "obstacletype.h"
-#include <vector>
-
-// LaserScan
-float angle_min; // minimum angle possible
```
Comment (airfield20) [inline src/obstacle_detection.cpp:7]: no global variables...

File: src/obstacle_detection.cpp [hunk 5]
```diff
@@ -1,43 +0,0 @@
...
-float angle_max; // maximum angle possible
-float angle_increment; // difference in angle for each reading
-float range_min; // minimum value of ranges
-float range_max; // maximum value of ranges
-float* ranges; // list of distance 'r' from each measured point
```
Comment (airfield20) [inline src/obstacle_detection.cpp:12]: Dont ever do this. Most segfaults are caused by indexing issues. 
For arrays use std::array E.g.
std::array ranges;
to get/set values use at()
ranges.at(0) = 5;
int var = ranges.at(3);

File: src/obstacle_detection.cpp [hunk 6]
```diff
@@ -1,43 +0,0 @@
...
-int end_index;
-
-/**
-    Called when a new message is published on sensor_msgs/LaserScan
-*/
-void laserScanCallback(const sensor_msgs::LaserScan::ConstPtr& scannedData) {
-    angle_min = scannedData->angle_min;
-    angle_max = scannedData->angle_max;
-    angle_increment = scannedData->angle_increment;
-
```
Comment (airfield20) [inline src/obstacle_detection.cpp:24]: All of this stuff should be in class, Ill talk to you guys about it tomorrow

File: src/obstacle_detection.cpp [hunk 7]
```diff
@@ -1,43 +0,0 @@
-#include "geometry_msgs/Point.h"
```
Comment (airfield20) [inline src/obstacle_detection.cpp:1]: Lets try not to pull things that don't build. I havent tested this version, but just as a general rule

File: include/obstacle_detection/obstacle.h [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+             *
+             * @param size new size of the obstacle
+             */
+            void set_size(const double &size);
+
+            /**
+             *
+             * @param interior_points new points for the obstacle
+             */
+            void set_interior_points(const std::vector<Point2D> interior_points);
```
Comment (emmaskychuks) [inline include/obstacle_detection/obstacle.h:88]: Pass by reference

File: include/obstacle_detection/point.h [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+             * @param x_in double representing x coordinate
+             * @param y_in double representing y coordinate
+             */
+            Point2D(const double &x_in, const double &y_in);
+
+            /**
+             * @brief method that constructs a geometry_msgs::Point for ros compatability
+             * @return a ros compatible point with z = 0
+             */
+            geometry_msgs::Point to_ros_point() const;
```
Comment (emmaskychuks) [inline include/obstacle_detection/point.h:33]: Convert to Point32

File: include/obstacle_detection/point.h [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+         * The purpose of this function is to give the ability to compare points that represent sensor readings
+            * No two sensor readings will be the same in continuous space therefore this thresholding function is necesary
+            * to compare points that are equal....but not quite
+         *
+         * @param p1 first ponit to compare
+         * @param p2 second point to compare
+         * @param threshold p1 and p2 must be within this value for them to be considered equal
+         * @return true is difference between points is less than threshold. Otherwise, false
+         */
+        bool close_enough(Point2D p1, Point2D p2, double threshold);
```
Comment (emmaskychuks) [inline include/obstacle_detection/point.h:69]: All params should be const references

File: include/obstacle_detection/boundary.h [hunk 11]
```diff
@@ -0,0 +1,70 @@
+#ifndef BOUNDARY_H
+#define BOUNDARY_H
+
+#include "point.h"
+
+namespace ISC {
+    namespace geometry {
+
+        class RectangleBoundary {
```
Comment (emmaskychuks) [inline include/obstacle_detection/boundary.h:9]: Change RectangleBoundary2D

**Pull request merged**